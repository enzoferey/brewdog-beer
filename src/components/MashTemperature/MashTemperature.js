import React from "react";
import PropTypes from "prop-types";
import style from "./mash-temperature.scss";

import temperatureIcon from "components/Table/temperature.png";

import getBeautifulTemp from "utils/getBeautifulTemp";

import { IconLabel } from "components/TableRow";

const MashTemperature = ({ index, value, unit, duration }) => (
  <div className={style.main}>
    <div className={style.indexContainer}>
      <span>{index}</span>
    </div>
    <IconLabel icon={temperatureIcon} label={getBeautifulTemp(value, unit)} />
    <span className={style.duration}>{`${duration}s`}</span>
  </div>
);

MashTemperature.propTypes = {
  index: PropTypes.number,
  value: PropTypes.number,
  unit: PropTypes.string,
  duration: PropTypes.number,
};

export default MashTemperature;
