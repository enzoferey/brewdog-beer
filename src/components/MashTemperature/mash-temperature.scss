$circle-index-size: 23px;

.main {
  flex: 1;
  display: flex;
  align-items: center;
}

.indexContainer {
  width: $circle-index-size;
  height: $circle-index-size;
  border-radius: 50%;
  background-color: #333;
  color: white;
  font-size: 14px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  margin-right: 15px;
}

.duration {
  border: 1px solid #333333;
  border-radius: 50px;
  padding: 2px 5px;
  margin-left: 10px;
  font-size: 14px;
  position: relative;
  top: 1px;
}
