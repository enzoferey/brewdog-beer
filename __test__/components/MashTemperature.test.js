import React from "react";
import MashTemperature from "components/MashTemperature";

const props = {
  index: 1,
  value: 19,
  unit: "celsius",
};

const propsDuration = {
  ...props,
  duration: 10,
};

test("renders correctly", () => {
  testSnapshot(<MashTemperature {...propsDuration} />);
});

test("doesn't display a duration when no `duration` props", () => {
  const wrapper = shallow(<MashTemperature {...props} />);
  expect(wrapper.children().length).toBe(2);
});
