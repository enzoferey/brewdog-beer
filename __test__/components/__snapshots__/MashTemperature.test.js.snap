// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <span>
      1
    </span>
  </div>
  <img
    alt=""
    src="temperature.png"
  />
  <span>
    19ÂºC
  </span>
  <span>
    10s
  </span>
</div>
`;
